### PR TITLE
Allow negative one for dynamic multiply

### PR DESCRIPTION
**Pull request by michalkulakowski**

### CODE REVIEW

File: src/pipelinedefinition.cpp [hunk 1]
```diff
@@ -479,7 +479,7 @@ class NodeValidator {
                 shape.size());
             return StatusCode::PIPELINE_NOT_ENOUGH_SHAPE_DIMENSIONS_TO_DEMULTIPLY;
         }
-        if (demultiplicatorNodeInfo.demultiplyCount.value() != 0) {
+        if (demultiplicatorNodeInfo.demultiplyCount.value() != 0 && demultiplicatorNodeInfo.demultiplyCount.value() != -1) {
             // TODO -1 vs 0 as demultiplexing value
```
Comment (atobiszei) [inline src/pipelinedefinition.cpp:483]: To remove

File: src/test/schema_test.cpp [hunk 2]
```diff
@@ -1419,14 +1419,25 @@ TEST(SchemaTest, DemultiplexerConfigMatchingSchema) {
     EXPECT_EQ(result, ovms::StatusCode::OK);
 }
 
-TEST(SchemaTest, DemultiplexerConfigDemultiplyCountNegativeNotAllowed) {
```
Comment (atobiszei) [inline src/test/schema_test.cpp:1422]: Do we have positive test for 0 as well?

**Review by dkalinowski — CHANGES_REQUESTED** [review]
There would be 90% less amount of changes required if we kept using 0 as dynamic demultiplyCount and only converted -1 to 0 at config reading time. This would also only require 1 more functional test that given config with -1 to works as well.

**Comment by dkalinowski** [comment]
There was also suggestion that we accepted `auto` keyword as well. Do we want to support it? @atobiszei @dtrawins

**Review by dkalinowski — APPROVED** [review]

**Review by atobiszei — APPROVED** [review]

**Pull request merged**